nilearn/image/image.py

```diff
@@ -596,3 +599,99 @@ def new_img_like(ref_niimg, data, affine=None, copy_header=False):
     return ref_niimg.__class__(data, affine, header=header)
+
+
+def threshold_img(img, threshold, mask_img=None,
+                  thresholding_strategy='img_value'):
+    """ A function keeps the most prominent regions of the input image,
+    also can be called as foreground extraction.
+
+    Parameters
+    ----------
+    img: a 3D/4D Nifti like image/object
+        a nifti image consists of statistical or atlas maps which needs
+        to be used in the thresholding process.
+
+    threshold: a float or a string or a number
+        A value used to threshold the image.
+        If given as float, this intensity value will be directly used to
+        threshold the image. The value should be within the range of minimum
+        intensity and maximum intensity of the given image.
+        Mostly suitable, if you exactly know which part of the brain regions
+        are to be kept. This value is used in thresholding_strategy='img_value'.
+        or
+        If given as string, it should finish with percent sign e.g. "80%" and
+        should be within the range of "0%" to "100%".
+        If given as number, it should be a real number of range between 0 and 100.
+        Both string and real number are used in thresholding_strategy='percentile'.
+
+    thresholding_strategy: string {'percentile', 'img_value'}, \
+        default 'percentile', optional
+        A strategy which takes the value and thresholds the image.
+        If 'percentile', image is thresholded based on the percentage of the
+        score on the image data. The scores which are survived above this
+        percentile are kept.
+        or
+        If 'img_value', voxels which have intensities greater than given float
+        value are kept.
+
+    mask_img: a Nifti like image/object, default is None, optional
+        A mask image used to mask the input image data. For example, if mask image
+        of one particular brain region is given then only those region will be
+        used for processing and rest of the regions are masked out to zero.
+        If None, no masking will be applied. Processing will be done on whole brain.
+
+    Returns
+    -------
+    threshold_img: a Nifti like image/object
+        a thresholded image of the given input image.
+    """
+    from . import resampling
+    from .. import masking
+
+    img = check_niimg(img, atleast_4d=True)
+    img_data = img.get_data()
+    affine = img.get_affine()
+
+    if mask_img is not None:
+        if not _check_same_fov(img, mask_img):
+            mask_img = resampling.resample_img(mask_img, target_affine=affine,
+                                               target_shape=img.shape[:3],
+                                               interpolation="nearest")
+
+        mask_data, _ = masking._load_mask_img(mask_img)
+        # Set as 0 for the values which are outside of the mask
+        img_data[mask_data == 0.] = 0.
+
+    if threshold is not None:
+        if isinstance(threshold, float):
```

Comment:
I think that we should not raise an error if the maps returned are empty: garbage in -> garbage out. We should maybe raise a warning but it has to be done in `check_threshold`. This whole block should be replaced by `threshold = check_threshold(...)`.